hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/file/OMFileRequest.java

```diff
@@ -64,24 +72,79 @@ public static OMDirectoryResult verifyFilesInPath(
         // Found a file in the given path.
         // Check if this is actual file or a file in the given path
         if (dbKeyName.equals(fileNameFromDetails)) {
-          return OMDirectoryResult.FILE_EXISTS;
+          result = OMDirectoryResult.FILE_EXISTS;
         } else {
-          return OMDirectoryResult.FILE_EXISTS_IN_GIVENPATH;
+          result = OMDirectoryResult.FILE_EXISTS_IN_GIVENPATH;
         }
       } else if (omMetadataManager.getKeyTable().isExist(dbDirKeyName)) {
         // Found a directory in the given path.
         // Check if this is actual directory or a directory in the given path
         if (dbDirKeyName.equals(dirNameFromDetails)) {
-          return OMDirectoryResult.DIRECTORY_EXISTS;
+          result = OMDirectoryResult.DIRECTORY_EXISTS;
         } else {
-          return OMDirectoryResult.DIRECTORY_EXISTS_IN_GIVENPATH;
+          result = OMDirectoryResult.DIRECTORY_EXISTS_IN_GIVENPATH;
+        }
+      } else {
+        if (!dbDirKeyName.equals(dirNameFromDetails)) {
+          missing.add(keyPath.toString());
         }
       }
+
+      if (result != OMDirectoryResult.NONE) {
+        return new OMPathInfo(missing, result);
+      }
       keyPath = keyPath.getParent();
     }
 
     // Found no files/ directories in the given path.
-    return OMDirectoryResult.NONE;
+    return new OMPathInfo(missing, OMDirectoryResult.NONE);
+  }
+
+  /**
+   * Get the valid base object id given the transaction id.
+   * @param id of the transaction
+   * @return base object id allocated against the transaction
+   */
+  public static long getObjIDFromTxId(long id) {
+    return getObjIdRangeFromTxId(id).getLeft();
+  }
```

Comment:
We use this function multiple times (more than getObjIdRangeFromTxId()). Wouldn't it be better to directly execute shift operation here instead of calling getObjIdRangeFromTxId()?

Reply:
ok